components/raftstore/src/store/async_io/write.rs

```diff
@@ -539,15 +525,22 @@ where
         STORE_WRITE_SEND_DURATION_HISTOGRAM
             .observe(duration_to_sec(now2.saturating_duration_since(now)));
 
-        for (region_id, (peer_id, ready_number)) in &self.batch.readies {
-            self.notifier
-                .notify_persisted(*region_id, *peer_id, *ready_number, now2);
+        if notify {
+            for (region_id, (peer_id, ready_number)) in &self.batch.readies {
+                self.notifier
+                    .notify_persisted(*region_id, *peer_id, *ready_number);
+            }
+            STORE_WRITE_CALLBACK_DURATION_HISTOGRAM
+                .observe(duration_to_sec(now2.saturating_elapsed()));
         }
-        STORE_WRITE_CALLBACK_DURATION_HISTOGRAM.observe(duration_to_sec(now2.saturating_elapsed()));
 
-        self.batch.clear();
```

Comment:
It's very easy to make mistake if batch is not cleared after written. How about clearing the batch and returning all readiness in `write_to_db`?

Reply:
I am thinking about completely removing the sync process.
I have added a new config `cmd-batch-concurrent-ready-max-count` (default 1), which means the proposal will be batched if the number of concurrent readies is greater than this config. By the way, I remove `raft-ready-size-limit` as well for better batching.
I do a test and the result is very good. The CPU is lower than previous while the QPS is better.
Although this batch pattern is not perfect, it's very similiar to original code. Maybe delaying a adaptive time is better but it's little hard to implement it efficiently.

Reply:
The switch can still be useful in extreme case if async IO is not working as expected. Now that it's implemented, I suggest to keep it, though it can be set as async by default.

Reply:
Got it.